query-node/mappings/src/workingGroups.ts

```diff
@@ -509,6 +526,12 @@ export async function workingGroups_OpeningFilled({ store, event }: EventContext
                 application.id
               )
             }
+
+            if (opening.type === WorkingGroupOpeningType.LEADER) {
+              // setting isLead of existing leader to false.
+              await removeIsLeadFromGroup(store, group.id)
```

Comment:
It's a detail but on line currently `566` there is:
```ts
// Update working group and LeaderSetEvent if necessary
if (opening.type === WorkingGroupOpeningType.LEADER && hiredWorkers.length) {
  group.leader = hiredWorkers[0]
  await store.save<WorkingGroup>(group)
  ...
```

IMO this:
```ts
if (opening.type === WorkingGroupOpeningType.LEADER) {
  // setting isLead of existing leader to false.
  await removeIsLeadFromGroup(store, group.id)
}
```
Should be called just before line `566` instead of in the `hiredWorkers` map. This way would be clearer that it's only called once.